x/slashing/src/keys.rs

```diff
@@ -0,0 +1,40 @@
+use crate::keeper::*;
+use gears::types::address::ConsAddress;
+
+pub(crate) fn addr_pubkey_relation_key(addr: ConsAddress) -> Vec<u8> {
+    let mut key = ADDR_PUBKEY_RELATION_KEY_PREFIX.to_vec();
+    let addr_bytes = Vec::from(addr);
+    let postfix = must_length_prefixed(&addr_bytes);
+    key.extend_from_slice(&postfix);
+    key
+}
+
+pub(crate) fn validator_signing_info_key(addr: ConsAddress) -> Vec<u8> {
+    let mut key = VALIDATOR_SIGNING_INFO_KEY_PREFIX.to_vec();
+    let addr_bytes = Vec::from(addr);
+    let postfix = must_length_prefixed(&addr_bytes);
+    key.extend_from_slice(&postfix);
+    key
+}
+
+pub(crate) fn validator_missed_block_bit_array_key(addr: ConsAddress, index: i64) -> Vec<u8> {
+    let mut key = VALIDATOR_MISSED_BLOCK_BIT_ARRAY_KEY_PREFIX.to_vec();
+
+    let addr_bytes = Vec::from(addr);
+    let index_bytes = index.to_le_bytes();
+    let postfix = must_length_prefixed(&addr_bytes);
+
+    key.extend_from_slice(&postfix);
+    key.extend_from_slice(&index_bytes);
+    key
+}
+
+fn must_length_prefixed(bytes: &[u8]) -> Vec<u8> {
```

Comment:
Can this accept a `ConsAddress`? Then you don't need the length checks.

Reply:
updated